index.js

```diff
@@ -1,72 +1,79 @@
-var path = require('path'),
-  fs = require('fs'),
-  jade = require('jade'),
-  url = require('url'),
-  assert = require('assert');
-
-var defaultOptions = {
-  // Jade options
-  jade: {},
-  // Serve index.jade if http://someurl/example/ is requested
+const path = require('path');
+const fs = require('fs'); // eslint-disable-line id-length
+const pug = require('pug');
+const url = require('url');
+const assert = require('assert');
+
+const defaultOptions = {
+  // Pug options
+  pug: {},
+  // Serve index.pug if http://someurl/example/ is requested
   serveIndex: true,
-  // Valid jade template extensions
-  ext: ['.jade'],
+  // Valid pug template extensions
+  ext: ['.pug'],
   // Allowed request extension
-  allowedExt: ['.jade', '.htm', '.html'],
+  allowedExt: ['.pug', '.htm', '.html'],
   // Header for Cache-Control: max-age=0
-  maxAge: 0
+  maxAge: 0,
 };
 
-module.exports = function(opts) {
 
-  if (!opts.baseDir)
+module.exports = function (opts) {
+  if (!opts.baseDir) {
     throw new Error('baseDir should be set');
+  }
 
-  if (!opts.baseUrl)
+  if (!opts.baseUrl) {
     throw new Error('baseUrl should be set');
+  }
 
-  opts = module.exports.getDefaultOptions(opts);
-
-  return function(req, res, next) {
+  const newOpts = module.exports.getDefaultOptions(opts);
 
-    if (req.originalUrl.indexOf(opts.baseUrl) !== 0)
+  return function (req, res, next) {
+    if (req.originalUrl.indexOf(newOpts.baseUrl) !== 0) {
       return next();
+    }
 
-    var filepath = module.exports.getTplPath(req.originalUrl, opts);
+    const filePath = module.exports.getTplPath(req.originalUrl, newOpts);
 
-    if (!filepath)
+    if (!filePath) {
       return next();
+    }
 
-    if (filepath.indexOf(opts.baseDir) !== 0)
+    if (filePath.indexOf(newOpts.baseDir) !== 0) {
       return res.sendStatus(403);
+    }
 
-    fs.stat(filepath, function(err, stats) {
-      if (err)
+    fs.stat(filePath, (err, stats) => {
+      if (err) {
         return next(err);
+      }
 
-      if (!stats.isFile())
-        return next();
-
-      jade.renderFile(filepath, opts.jade, function renderFile(err, html) {
-        if (err)
-          return next(err);
-
-        res.setHeader('Content-Length', Buffer.byteLength(html));
-        res.setHeader('Content-Type', 'text/html; charset=utf-8');
-        var now = +new Date();
-        var expires = new Date(now + opts.maxAge);
-        res.setHeader('Cache-Control', 'max-age=' + opts.maxAge);
-        res.setHeader('Expires', expires.toGMTString());
-        return res.end(html);
-      });
-    });
-  };
+      if (!stats.isFile()) {
+      return next();
+    }
 
+    pug.renderFile(filePath, newOpts.pug, (renderErr, html) => {
+      if (renderErr) {
+        return next(renderErr);
+      }
+
+    res.setHeader('Content-Length', Buffer.byteLength(html));
+    res.setHeader('Content-Type', 'text/html; charset=utf-8');
+    const now = +new Date();
```

Comment:
might be confusing as `now` in timestamp while `expires` as Date. maybe rename variable?